modules/apps/portal-security/portal-security-password-encryptor-impl/src/main/java/com/liferay/portal/security/password/encryptor/internal/PBKDF2PasswordEncryptor.java

```diff
@@ -134,6 +139,23 @@ public void configure(String algorithm, String encryptedPassword)
 						bufferUnderflowException);
 				}
 			}
+
+			int indexOf = StringUtil.indexOfIgnoreCase(algorithm, "SHA");
+
+			if (indexOf < 0) {
+				return;
+			}
+
+			_digest = DigestFactory.getDigest(
+				algorithm.substring(
```

Comment:
@marianoalvarosaiz please remove the chaining here. After that, we can forward this. Thanks for applying the new approach.